cpp/CMakeLists.txt

```diff
@@ -246,57 +250,72 @@ file(INSTALL ${LIBCXX_INCLUDE_DIR}     DESTINATION ${CMAKE_BINARY_DIR}/include/l
+message(STATUS "ZLIB: ZLIB_LIBRARY set to ${ZLIB_LIBRARY}")
+message(STATUS "ZLIB: ZLIB_INCLUDE_DIR set to ${ZLIB_INCLUDE_DIR}")
 
-if(ZLIB_FOUND)
-    message(STATUS "ZLib found in ${ZLIB_INCLUDE_DIRS}")
+if(ZLIB_INCLUDE_DIR)
+    message(STATUS "ZLib found in ${ZLIB_INCLUDE_DIR}")
 else()
     message(FATAL_ERROR "ZLib not found, please check your settings.")
-endif(ZLIB_FOUND)
+endif(ZLIB_INCLUDE_DIR)
+
 
 ###################################################################################################
 # - find boost ------------------------------------------------------------------------------------
 
 # Don't look for a CMake configuration file
 set(Boost_NO_BOOST_CMAKE ON)
 
-find_package(
-    Boost REQUIRED MODULE
-    COMPONENTS filesystem
+CPMFindPackage(
+  NAME Boost
+  GITHUB_REPOSITORY Orphis/boost-cmake@60fb5fa
+  VERSION 1.70.0
+  FIND_PACKAGE_ARGUMENTS "COMPONENTS filesystem"
 )
 
-message(STATUS "BOOST: Boost_LIBRARIES set to ${Boost_LIBRARIES}")
-message(STATUS "BOOST: Boost_INCLUDE_DIRS set to ${Boost_INCLUDE_DIRS}")
+message(STATUS "BOOST: Boost_LIBRARY set to ${Boost_LIBRARY}")
+message(STATUS "BOOST: Boost_INCLUDE_DIR set to ${Boost_INCLUDE_DIR}")
 
-if(Boost_FOUND)
-    message(STATUS "Boost found in ${Boost_INCLUDE_DIRS}")
+if(Boost_INCLUDE_DIR)
+    message(STATUS "Boost found in ${Boost_INCLUDE_DIR}")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DBOOST_NO_CXX14_CONSTEXPR")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_NO_CXX14_CONSTEXPR")
     set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -DBOOST_NO_CXX14_CONSTEXPR")
 else()
     message(FATAL_ERROR "Boost not found, please check your settings.")
-endif(Boost_FOUND)
+endif(Boost_INCLUDE_DIR)
 
 ###################################################################################################
 # - RMM -------------------------------------------------------------------------------------------
 
-find_path(RMM_INCLUDE "rmm"
-          HINTS "$ENV{RMM_ROOT}/include")
+# FindCUDA doesn't quite set the variables it should. RMM needs this to be set to build from CPM
+set(CUDAToolkit_INCLUDE_DIR "${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}")
+
+CPMFindPackage(
+  NAME RMM
+  GITHUB_REPOSITORY rapidsai/rmm
+  VERSION ${CMAKE_PROJECT_VERSION}
+  GIT_TAG "branch-${CMAKE_PROJECT_VERSION_MAJOR}.${CMAKE_PROJECT_VERSION_MINOR}"
+)
 
-message(STATUS "RMM: RMM_INCLUDE set to ${RMM_INCLUDE}")
+message(STATUS "RMM: RMM_SOURCE_DIR set to ${RMM_SOURCE_DIR}")
 
 ###################################################################################################
 # - DLPACK -------------------------------------------------------------------------------------------
 
-find_path(
-    DLPACK_INCLUDE "dlpack"
-    HINTS "$ENV{DLPACK_ROOT}/include"
+CPMFindPackage(
+  NAME DLPACK
+  GITHUB_REPOSITORY dmlc/dlpack
+  VERSION 0.3
 )
 
-message(STATUS "DLPACK: DLPACK_INCLUDE set to ${DLPACK_INCLUDE}")
+message(STATUS "DLPACK: dlpack_SOURCE_DIR set to ${dlpack_SOURCE_DIR}")
```

Comment:
If dlpack is found I imagine the `SOURCE_DIR` wouldn't be set?

Reply:
DLPACK_INCLUDE is no longer needed. In both the finding system and installing from source this appears to be set with CPM. I'm not saying that's right. Maybe another message is a better idea? I could just do a check for `dlpack_FOUND` just so the logs are descriptive? If found, STATUS "dlpack FOUND", else "source_dir"?

Reply:
We'd get a `dlpack` target that we could use instead of variables like this regardless, right?